kotlin-extensions/src/main/java/com/amplifyframework/ktx/DataStoreCategoryExt.kt

```diff
@@ -0,0 +1,157 @@
+import com.amplifyframework.core.model.query.QueryOptions
+import com.amplifyframework.core.model.query.predicate.QueryPredicate
+import com.amplifyframework.datastore.DataStoreCategory
+import com.amplifyframework.datastore.DataStoreException
+import com.amplifyframework.datastore.DataStoreItemChange
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.channels.sendBlocking
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+@Throws(DataStoreException::class)
+suspend fun <T: Model> DataStoreCategory.save(item: T) {
+    return suspendCoroutine { continuation ->
+        save(item,
+            { continuation.resume(Unit) },
+            { continuation.resumeWithException(it) }
+        )
+    }
+}
+
+@Throws(DataStoreException::class)
+suspend fun <T: Model> DataStoreCategory.save(item: T, predicate: QueryPredicate) {
+    return suspendCoroutine { continuation ->
+        save(item, predicate,
+            { continuation.resume(Unit) },
+            { continuation.resumeWithException(it) }
+        )
+    }
+}
+
+@Throws(DataStoreException::class)
+suspend fun <T: Model> DataStoreCategory.delete(item: T) {
+    return suspendCoroutine { continuation ->
+        delete(item,
+            { continuation.resume(Unit) },
+            { continuation.resumeWithException(it) }
+        )
+    }
+}
+
+@Throws(DataStoreException::class)
+suspend fun <T: Model> DataStoreCategory.delete(item: T, predicate: QueryPredicate) {
+    return suspendCoroutine { continuation ->
+        delete(item, predicate,
+            { continuation.resume(Unit) },
+            { continuation.resumeWithException(it) }
+        )
+    }
+}
+
+@Throws(DataStoreException::class)
+suspend fun <T: Model> DataStoreCategory.query(clazz: Class<T>): Iterator<T> {
+    return suspendCoroutine { continuation ->
+        query(
+            clazz,
+            { continuation.resume(it) },
+            { continuation.resumeWithException(it) }
+        )
+    }
+}
+
+@Throws(DataStoreException::class)
+suspend fun <T: Model> DataStoreCategory.query(clazz: Class<T>, predicate: QueryPredicate): Iterator<T> {
+    return suspendCoroutine { continuation ->
+        query(
+            clazz,
+            predicate,
+            { continuation.resume(it) },
+            { continuation.resumeWithException(it) }
+        )
+    }
+}
+
+@Throws(DataStoreException::class)
+suspend fun <T: Model> DataStoreCategory.query(clazz: Class<T>, options: QueryOptions): Iterator<T> {
+    return suspendCoroutine { continuation ->
+        query(
+            clazz,
+            options,
+            { continuation.resume(it) },
+            { continuation.resumeWithException(it) }
+        )
+    }
+}
+
+// TOOD: Flow is from kotlinx, which requires another dependency,
```

Comment:
This is fine. If you're using coroutines you're already depending on the `kotlinx-coroutines-core`